### PR TITLE
ci: migrate legacy saucelabs job to Github Actions

### PR DESCRIPTION
**Pull request by josephperrott**
Migrate the legacy saucelabs job

### CODE REVIEW

**Comment by devversion** [comment]
FYI: The job is failing and I also wonder about this comment:

```
      # In order to avoid the bottleneck of having a slow host machine, we acquire a better
      # container for this job. This is necessary because we launch a lot of browsers concurrently
      # and therefore the tunnel and Karma need to process a lot of file requests and tests.
```

I just feel like we are discarding many of the past learnings simply as part of this migration. In my opinion, we should rather keep things as is, perform the migration, and then consider downgrading runner size in follow-ups with more data. Sorry for bringing up that point again.

**Review by devversion — COMMENTED** [review]
(see comment above)

**Comment by josephperrott** [comment]
@devversion 
Granted this is a limited sample size but here are the timings of this specific job using different runners (notably this job does not rely on bazel, so caching is not involved here in the same way):

| Runner Size | Overall Time | Test Step Time |
| ----------- | ------------ | -------------- |
|    2 Core   |    8m 1s     |     2m 6s      |
|    4 Core   |    6m 5s     |     2m 15s     |
|    8 Core   |    6m 31s    |     2m 8s      |
|   16 Core   |    5m 49s    |     1m 58s     |
|   32 Core   |    6m 7s     |     2m 18s     |
|   64 Core   |    6m 20s    |     2m 12s     |


From looking at this, I think that we get diminishing returns pretty quickly on the larger runner size.  I agree that we should be sure to have good sized runners, but I also want to avoid overpaying just because we used large runners in the past.

**Comment by devversion** [comment]
> From looking at this, I think that we get diminishing returns pretty quickly on the larger runner size. I agree that we should be sure to have good sized runners, but I also want to avoid overpaying just because we used large runners in the past.

For sure, but we need some numbers- especially if we had a comment with explicit reasoning on why we used a larger runner in the past.  

Thanks for collecting these numbers. I agree that 4 cores seems ideal here?

File: .github/workflows/ci-privileged.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    env:
+      SAUCE_TUNNEL_IDENTIFIER: angular-framework-${{ github.run_number }}
+      SAUCE_USERNAME: ${{ vars.SAUCE_USERNAME }}
+      SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+    steps:
+      - name: Initialize environment
+        uses: angular/dev-infra/github-actions/npm/checkout-and-setup-node@1befe6d24fd5cab818e7f8bada40ab364781b8a8
+        with:
+          cache-node-modules: true
+          ref: ${{ github.sha }}
```
Comment (devversion) [inline .github/workflows/ci-privileged.yml:29]: One thing: Do we want to add a comment for every job in the priviledged file and clarify here that checking out the fork user code is _intended_ (in terms of access of these secrets)

**Review by devversion — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit d91618f2d81eec5f35dcd00d1f31e80fd9aace1a.

**Pull request closed without merging**